ietf-optical-impairment-topology.yang

```diff
@@ -1292,30 +1303,95 @@ Concentratedloss)";
+       }
 
-         choice mode {
+       list tranceivers-list {
+         key "tranceiver-id";
+         config false;
+         description "list of tranceivers related to a transponder";
+         leaf tranceiver-id {
+           type uint32;
+           description "tranceiver identifier";
+         }
+   
+         list supported-modes {
+           key "mode-id";
+           config false;
+           description "list of capabilities for tranceiver";
+           leaf mode-id {
+             type string { 
+               length "1..255";
+             }
+             description "ID for supported tranceiver 
+             capabilities";  
+           }
+           choice mode {
+             mandatory true;
              description "standard mode, organizational mode or
-              explicit mode";
+                          explicit mode";
 
              case G.692.2 {
                uses standard-mode;
              }
 
              case organizational-mode {
                uses organizational-mode;
+
+/*add "common-tranceiver-capabilities-organizational-mode" 
+from layer0-ext*/
+/*add "common-tranceiver-capabilities-organizational-explicit"
+from layer0-ext*/
+
              }
 
              case explicit-mode {
-               uses transponder-attributes;
-             }
-         }
+               container supported-std-modes {
+                 description "container to group
+                  all supported modes";
+                 leaf-list supported-application-codes {
+                   type leafref {
+                    path "../../mode-id";
+                   }
+must "../../../supported-modes[mode-id=current()]/standard-mode"{
+                     description
+                       "the pointer is only for application-code 
+                         supported by tranceiver.";
+                   }
+
+                   description "list of 
+                      pointers to supported supported 
+                      application-code"; 
+                 }
+
+                 leaf-list supported-organizational-modes {
+                   type leafref {
+                    path "../../mode-id";
+                   }
+
+must "../../../supported-modes[mode-id=current()]/operational-mode"{
+                     description
+                       "the pointer is only for 
+                       organizational-mode supported
+                        by tranceiver.";                     
+                   }
+                   description "list of 
+                      pointers to supported organizational-mode";
+                 }
+               }  // end of container
+
+ // add "common-tranceiver-capabilities-organizational-explicit"
+ // from layer0-ext   
+
+             } // end of case explicit-mode
+           } // end of choice
+         } // end of list of supported modes for
+           //each tranceiver related to one transponder
 
          leaf power {
```

Comment:
Unit missing: dBm

Reply:
eaf power {
           type int32;
           units "dBm";
           config false;
           description "per channel power";
         }

What is missing here ?

Reply:
leaf power {
           type int32;
           units "dBm";
           config false;
           description "per channel power";
         }

What is missing?

Reply:
Hi Sergio, my fault, I missed the expand button for the code in github.